components/sso-hostobject/org.wso2.carbon.hostobjects.sso/src/main/java/org/wso2/carbon/hostobjects/sso/internal/builder/AuthReqBuilder.java

```diff
@@ -60,26 +73,92 @@ public AuthnRequest buildAuthenticationRequest(String issuerId) throws Exception
      */
     public AuthnRequest buildPassiveAuthenticationRequest(String issuerId, String acsUrl) throws Exception  {
         Util.doBootstrap();
-        //matches shortest segments that are between '{' and '}'
-        Pattern pattern = Pattern.compile("\\$\\{(.*?)\\}");
-        Matcher matcher = pattern.matcher(acsUrl);
-        while (matcher.find()) {
-            String match = matcher.group(1);
-            String property = System.getProperty(match);
-            if (property != null) {
-                acsUrl = acsUrl.replace("${" + match + "}", property);
-            } else {
-                log.warn("System Property " + match + " is not set");
-            }
-        }
+        acsUrl = processAcsUrl(acsUrl);
         AuthnRequest authnRequest = (AuthnRequest) Util.buildXMLObject(AuthnRequest.DEFAULT_ELEMENT_NAME);
         authnRequest.setID(Util.createID());
         authnRequest.setVersion(SAMLVersion.VERSION_20);
         authnRequest.setIssueInstant(new DateTime());
-        authnRequest.setIssuer(buildIssuer( issuerId));
+        authnRequest.setIssuer(buildIssuer(issuerId));
+        authnRequest.setNameIDPolicy(buildNameIDPolicy());
+        authnRequest.setIsPassive(true);
+        authnRequest.setAssertionConsumerServiceURL(acsUrl);
+        return authnRequest;
+    }
+
+    /**
+     * Generate a signed Authentication request with passiveAuth
+     *
+     * @param issuerId
+     * @param acsUrl
+     * @return
+     * @throws Exception
+     */
+    public AuthnRequest buildPassiveSignedAuthenticationRequest(String issuerId, int tenantId,
+            String tenantDomain, String destination, String acsUrl) throws Exception {
+        Util.doBootstrap();
+        acsUrl = processAcsUrl(acsUrl);
+        AuthnRequest authnRequest = (AuthnRequest) Util.buildXMLObject(AuthnRequest.DEFAULT_ELEMENT_NAME);
+        authnRequest.setID(Util.createID());
+        authnRequest.setVersion(SAMLVersion.VERSION_20);
+        authnRequest.setIssueInstant(new DateTime());
+        authnRequest.setIssuer(buildIssuer(issuerId));
         authnRequest.setNameIDPolicy(buildNameIDPolicy());
         authnRequest.setIsPassive(true);
         authnRequest.setAssertionConsumerServiceURL(acsUrl);
+        authnRequest.setDestination(destination);
+        SSOAgentCarbonX509Credential ssoAgentCarbonX509Credential =
+                new SSOAgentCarbonX509Credential(tenantId, tenantDomain);
+        setSignature(authnRequest, SignatureConstants.ALGO_ID_SIGNATURE_RSA,
+                new X509CredentialImpl(ssoAgentCarbonX509Credential));
+        return authnRequest;
+    }
+
+    /**
+     * Generate an signed authentication request.
+     *
+     * @return Signed AuthnRequest Object
+     * @throws Exception error when bootstrapping
+     */
+    public AuthnRequest buildSignedAuthRequest(String issuerId, int tenantId,
+            String tenantDomain) throws Exception {
```

Comment:
Fixed in #1528 (https://github.com/wso2/carbon-apimgt/commit/c6d6cb237bd7654eda952fb2fbd74863e9899cfc)